bin/fix_my_workspace

```diff
@@ -4,8 +4,17 @@ $VERBOSE = nil  # Don't care about world writable dir warnings and the like
 if $_fix_my_workspace_version_check
   $_fix_my_workspace_version = '0.5.0'
 else
-  require 'origen'
-
+  # if we're running this from a development environment (i.e., bin/ inside of origen repo) then
```

Comment:
The fix_my_workspace script is owned/provided by the origen_updater plugin. The reason that it is not exposed as a regular shared command is that it has to be available at times when something has gone wrong with the user's workspace and they cannot invoke origen.

I went back and forth over a few different ways to guarantee that the user would always have access to the script, but in the end I felt the less magic the better and it was hard to beat just copying it into the application's bin directory.
That meant they can always just run ./bin/fix_my_workspace directly without having to go through Origen or Bundler's dispatch systems.

I still wanted to manage the script in a central location of course, so every time origen is invoked within an application that includes origen_updater, it will have a quick check to see if the version of the script they have locally is the one from the current version of origen_updater, if not it will copy it over on the fly.

Now ideally, I would prefer not to have required origen at all within that script. But since I needed access to the site_config to check for what system gems Origen should be using, I decided just to require it rather than try to re-implement the site config parsing within the script itself.
So far, that seems to have worked ok, but there are probably things you need to consider here with the changes to Origen's dispatch.
I don't really understand the changes well enough to make a recommendation, you will know best yourself, but just think about we need to maximise the chances of the script working even if an application's workspace is broken such that "origen" cannot be invoked through Bundler in the normal way.